### PR TITLE
Add Filter Feature

### PR DESCRIPTION
**Pull request by tanyqcollin**
User able to filter person list with specifying condition.
The filter command is filter by matching keywords.

**Main Changes**
- Added `FilterCommand` to filter the person list.
- Added `FilterCommandParser` to handle filter command input.
- Added `FilterPredicate` as a predicate to filter person list.
- Added relevant testcases for filter command.

> [!CAUTION]
> Each prefix (e.g., for name, phone, etc.) can only handle one word.
To manage multiple tags and modules, you can add multiple prefixes as needed.

### CODE REVIEW

**Review by RadieonAjax — APPROVED** [review]
LGTM!

**Review by yongqqqq — APPROVED** [review]
Overall good code. 

LGTM!

File: src/main/java/seedu/address/logic/commands/FilterCommand.java [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+            + "Parameters: KEYWORD [MORE_KEYWORDS]...\n"
+            + "Example: " + COMMAND_WORD + " n/joe m/MA1521";
+
+    private final FilterPersonDescriptor filterPersonDescriptor;
+    private final FilterPredicate predicate;
+
+    /**
+     * @param filterPersonDescriptor details to filter thr person with
+     */
+    public FilterCommand(FilterPersonDescriptor filterPersonDescriptor) {
```
Comment (Jianwen0451) [inline src/main/java/seedu/address/logic/commands/FilterCommand.java:42]: Maybe can consider move this static class out to form a separate class  in later iteration

File: src/main/java/seedu/address/logic/parser/FilterCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    /**
+     * Returns name if the name contains one word.
+     * @throws ParseException If the name contains more than one word.
+     */
+    private Name isNameValid(Name name) throws ParseException {
+        String[] words = name.fullName.split("\\s+");
+        if (words.length != 1) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FilterCommand.MESSAGE_USAGE));
+        }
+        return name;
```
Comment (Jianwen0451) [inline src/main/java/seedu/address/logic/parser/FilterCommandParser.java:112]: I think from line 76 to this line should be written in parserUtil

**Review by Jianwen0451 — APPROVED** [review]
The code looks good to me. Just mentioned a few changes that can make the code cleaner

File: src/main/java/seedu/address/logic/commands/FilterCommand.java (hunk 1 above)
Comment (tanyqcollin) [reply src/main/java/seedu/address/logic/commands/FilterCommand.java:42]: I refer to the `EditCommand` class, they did something similar like this

File: src/main/java/seedu/address/logic/parser/FilterCommandParser.java (hunk 2 above)
Comment (tanyqcollin) [reply src/main/java/seedu/address/logic/parser/FilterCommandParser.java:112]: noted

**Pull request merged**